ocs_ci/framework/__init__.py

```diff
@@ -479,6 +481,36 @@ def __init__(self):
             primary_index = primary_config.MULTICLUSTER.get("multicluster_index")
             super().__init__(primary_index)
 
+    class RunWithProviderConfigContext(RunWithConfigContext):
+        """
+        Context manager that makes sure that a given code block is executed on Provider.
```

Comment:
I would mention in docstring that if no provider cluster found, the code will run on current index.
And not sure we should change name of class to something like:
`RunWithProviderOrCurrnetConfigContext`

BTW This logic makes sense only if we would like to have this code using this context manager to work in provider/consumer as well as other type of executions, correct?

Reply:
> I would mention in docstring that if no provider cluster found, the code will run on current index.

This makes sense.

> And not sure we should change name of class to something like:
> RunWithProviderOrCurrnetConfigContext

How about `RunWithProviderConfigContextIfAvailable`?

> BTW This logic makes sense only if we would like to have this code using this context manager to work in provider/consumer as well as other type of executions, correct?

Yes, this way people can add it to the code without a worry that it breaks other test cases. It allows to mark chunks of code with a context for provider/consumer in generic function e.g. in `ocs_ci/ocs/resources/*` or `tests/conftest.py`